src/components/EventsDashboard/EventCard.tsx

```diff
@@ -0,0 +1,141 @@
+          REGISTER BY {`${deadline.getMonth() + 1}/${deadline.getDate()}`}
+        </div>
+      );
+    } else {
+      return <></>;
+    }
+  };
+
+  const registeredIndicator = (ev: BiztechEvent) => {
+    if (registered.includes(`${ev.id};${ev.year}`)) {
+      return <div className="rounded-full font-poppin font-[700] px-3 py-1 text-white bg-black text-[8px] lg:text-[12px]  flex items-center">REGISTERED</div>;
+    } else {
+      return <></>;
+    }
+  };
+
+  return (
+    <>
+      <Link href={`/event/${event.id}/${event.year}`}>
+        <motion.div
+          key={`${event.id + event.year + event.createdAt}`}
+          layout
+          initial={{ y: -10, opacity: 0 }}
+          animate={{ y: 0, opacity: 1 }}
+          exit={{ y: 100, opacity: 0 }}
+          transition={{
+            type: "tween",
+            ease: "easeInOut",
+            duration: 0.3,
+          }}
+          onClick={(e) => e.stopPropagation()}
+        >
+          <div className="bg-events-user-card-bg w-full p-3 rounded-[10px] my-2">
+            <div className="flex flex-row space-x-5 relative">
+              <div className="relative lg:w-[200px] lg:h-[130px] w-[100px] h-[75px] overflow-hidden rounded-lg shrink-0">
+                <Image className="object-cover" fill src={event.imageUrl} alt={event.ename} />
+              </div>
+              <div className="flex flex-col space-y-1 grow">
+                <div className="font-600 text-sm lg:text-[24px] py-0.5 lg:py-2 flex flex-row space-x-3 items-center w-full">
+                  <div>{event.ename}</div>
+                  <div className="grow"></div>
+                  <div className="hidden lg:block">{registeredIndicator(event)}</div>
+                  <div className="hidden lg:block">{timeStateIndicator(event)}</div>
+                  <div className="">
+                    <Bookmark
+                      height={30}
+                      width={30}
+                      onClick={(e) => handleSaveClick(e, event.id, event.year)}
+                      className={`cursor-pointer ${saved.includes(`${event.id};${event.year}`) || fill ? "fill-white" : ""}`}
+                    />
+                  </div>
+                </div>
+                <p className="text-[10px] lg:text-sm text-events-baby-blue">
+                  {`${months[dateString.getMonth()]} ${dateString.getDate()}, ${event.year}`}, {dateString.toTimeString().slice(0, 5)}
+                </p>
+                <div className="flex flex-row items-center justify-between w-full">
+                  <p className="text-[10px] lg:text-sm text-events-baby-blue">
+                    {`${event.pricing ? "$" + event.pricing.members.toFixed(2) : "Free!"}`}{" "}
+                    {event.pricing.nonMembers ? `(Non-members ${event.pricing?.nonMembers.toFixed(2)})` : "(Members only)"}
+                  </p>
```

Comment:
I think code readability here could be improved by Extracting Functions, similar to how you created `registeredIndicator` and `timeStateIndicator`- what do you think